src/components/forms/DateRangePicker/DateRangePicker.tsx

```diff
@@ -0,0 +1,145 @@
+import classnames from 'classnames'
+import { format } from 'prettier'
+import React, { useState } from 'react'
+import { DEFAULT_EXTERNAL_DATE_FORMAT, INTERNAL_DATE_FORMAT } from '../DatePicker/constants'
+import { DatePicker, DatePickerProps } from '../DatePicker/DatePicker'
+import { formatDate, parseDateString } from '../DatePicker/utils'
+
+interface DateRangePickerProps {
+  startDateLabel?: string
+  startDateHint?: string
+  startDatePickerProps: Omit<DatePickerProps, 'rangeDate'>
+  endDateLabel?: string
+  endDateHint?: string
+  endDatePickerProps: Omit<DatePickerProps, 'rangeDate'>
+  className?: string
+}
```

Comment:
I deviated slightly from what USWDS recommends here. It would be no big deal to change though, so I'm curious on reviewers' thoughts. USWDS recommends a min date and max date for the entire date range picker. At present, instead, this implementation allows a range to be specified on each date picker respectively. This would allow for the use case where the start date must fall within a given range, and the end date must fall within a separate range.

Once a start or end date is selected, however, the range of the other date picker is bounded by start/end date OR the specified date range, whichever is more restrictive.

Reply:
I think I see what you're saying but I need a little more clarification to understand fully.. are you saying the USWDS version would allow for incompatible start/end dates, and the way you did it will guard against that? if so I'm all in favor

Reply:
It wouldn't cause problems that we could adjust the logic for internally so much.

If we want this to match USWDS perfectly, we'd just expose a min and max date prop for the DateRangePicker, omit the min and max date pass through props for each DatePicker, and feed them in manually. 

The only reason I didn't do that is that the current solution allows this behavior (pass the min and max dates to each date picker), but also allows an implementer to specify different min and max dates for each date picker if they wanted to have separate min/max dates for each date picker. 

It just seemed like a feature that might be useful. I wanted to call it out and defer to reviewers opinions on that though. I can pretty easily refactor it to constrain the min and max date to apply to the entire date range picker rather than each date.

Reply:
Ahh I see, that makes sense! I like how you expanded it so I vote to keep it!